### PR TITLE
layers: Allow using 0th Queue Family and 1 queue

### PR DESCRIPTION
**Pull request by krOoze**
despite the strict VU wording, the counts can be infered:

> Implementations **must** support at least one queue family.

> Each queue family **must** support at least one queue.

### CODE REVIEW

**Comment by tobine** [comment]
Hi @krOoze. I think these changes are motivated/supported by the spec statement:

> Implementations must support at least one queue family.

Is that correct? The changes in general seem *ok* (and the positive test is fantastic!) but I think the same result can be achieved with less code defaulting the physical properties queue_family_count [here](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/layers/core_validation.h#L106) to "1" and defaulting queue_family_properties[0].queueCount right below it to "1".

Does that make sense and seem a bit cleaner? Any thoughts from others?

**Comment by krOoze** [comment]
@tobine I quoted the relevant language in the OP.

> can be achieved with less code defaulting the physical properties queue_family_count here to "1" and defaulting queue_family_properties[0].queueCount right below it to "1".

Perhaps, but it is somewhat "smarter" solution. I was to lazy to check if that would not have some unintended consequences.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -2363,7 +2363,7 @@ bool CoreChecks::ValidateDeviceQueueCreateInfos(const PHYSICAL_DEVICE_STATE *pd_
         }
 
         // Verify that requested  queue count of queue family is known to be valid at this point in time
-        if (requested_queue_family < pd_state->queue_family_count) {
+        if (requested_queue_family < pd_state->queue_family_count || requested_queue_family == 0) {
```
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp]: Maybe add a comment... `// Queue family 0 is guaranteed to exist by the spec` here and above.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -2376,8 +2376,9 @@ bool CoreChecks::ValidateDeviceQueueCreateInfos(const PHYSICAL_DEVICE_STATE *pd_
                     : "i.e. is not less than or equal to " +
                           std::to_string(pd_state->queue_family_properties[requested_queue_family].queueCount);
 
-            if (!queue_family_has_props ||
-                requested_queue_count > pd_state->queue_family_properties[requested_queue_family].queueCount) {
+            if (requested_queue_count > 1 &&
```
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp]: Similar comment that queue families must have at least one queue?

File: tests/vkpositivelayertests.cpp [hunk 3]
```diff
@@ -7185,3 +7185,28 @@ TEST_F(VkPositiveLayerTest, HostQueryResetSuccess) {
 
     m_errorMonitor->VerifyNotFound();
 }
+
+TEST_F(VkPositiveLayerTest, UseFirstQueueUnqueried) {
```
Comment (jzulauf-lunarg) [inline tests/vkpositivelayertests.cpp]: Clean.

**Comment by krOoze** [comment]
So I went somewhat with @tobine's suggestion, while being little bit more invasive. Please re-review.

**Comment by tobine** [comment]
Thank, @krOoze! I'm happy with this. Just needs clang-format.

**Comment by krOoze** [comment]
@tobine not anymore it doesn't

**Pull request merged**